playbooks/server-install.yml

```diff
@@ -135,3 +136,73 @@
         key: "{{ keys }}"
         state: present
         exclusive: true
+
+    - name: Create keyrings directory
+      file:
+        path: /etc/apt/keyrings
+        state: directory
+        mode: 0755
+
+    - name: Add Docker GPG apt Key
+      get_url:
+        url: https://download.docker.com/linux/debian/gpg
+        dest: /etc/apt/keyrings/docker.asc
+        mode: 0644
+
+    - name: Add Docker repository
+      apt_repository:
+        repo: "deb [arch=amd64 signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/debian {{ ansible_lsb.codename }} stable"
+        filename: docker
+        state: present
+
+    - name: Install docker
+      apt:
+        name:
+          - docker-ce
+          - docker-ce-cli
+          - containerd.io
+          - docker-buildx-plugin
+          - docker-compose-plugin
+        state: present
+        update_cache: yes
+
+    - name: Ensure /opt/nginx directory exists
+      file:
+        path: /opt/nginx
+        state: directory
+        owner: root
+        group: root
+        mode: 0755
+
+    - name: Create docker compose file
+      template:
+        src: ../templates/docker-compose.yaml.j2
+        dest: /opt/nginx/docker-compose.yaml
+        owner: root
+        group: root
+        mode: 0644
+
+    - name: Create nginx cache file
```

Comment:
can we update this to provisioner-cache? Basically remove the instances of nginx to provisioner-cache?